src/editors/editorsController.ts

```diff
@@ -229,117 +232,132 @@ export default class EditorsController {
     });
   }
 
-  saveDocumentToMongoDB(): Promise<boolean> {
```

Comment:
This method was causing a complexity lint error, this pr adds a `replaceDocument` function which does the MongoDB findOneAndReplace of it. cc @alenakhineika